MONAIAuto3DSeg/MONAIAuto3DSegLib/dependency_handler.py

```diff
@@ -0,0 +1,130 @@
+import sys
+
+import shutil
+import subprocess
+import logging
+
+from MONAIAuto3DSegLib.constants import APPLICATION_NAME
+logger = logging.getLogger(APPLICATION_NAME)
+
+
+from abc import ABC, abstractmethod
+
+
+class DependenciesBase(ABC):
+
+    minimumTorchVersion = "1.12"
+
+    def __init__(self):
+        self.dependenciesInstalled = False  # we don't know yet if dependencies have been installed
+
+    @abstractmethod
+    def installedMONAIPythonPackageInfo(self):
+        pass
+
+    @abstractmethod
+    def setupPythonRequirements(self, upgrade=False):
+        pass
+
+
+class LocalPythonDependencies(DependenciesBase):
+
+    def installedMONAIPythonPackageInfo(self):
+        versionInfo = subprocess.check_output([sys.executable, "-m", "pip", "show", "MONAI"]).decode()
+        return versionInfo
+
+    def _checkModuleInstalled(self, moduleName):
+      try:
+        import importlib
+        importlib.import_module(moduleName)
+        return True
+      except ModuleNotFoundError:
+        return False
+
+    def setupPythonRequirements(self, upgrade=False):
+        def install(package):
+          subprocess.check_call([sys.executable, "-m", "pip", "install", package])
+
+        logger.info("Initializing PyTorch...")
+
+        packageName = "torch"
+        if not self._checkModuleInstalled(packageName):
+          logger.info("PyTorch Python package is required. Installing... (it may take several minutes)")
+          install(packageName)
```

Comment:
This may have been misleading then. The dependency_handler's task is to differentiate between `LocalPythonDependencies' (running fastapi server locally) vs `SlicerPythonDependencies` (running webserver within 3D Slicer).

Reply:
Is PyTorch always running in Slicer's Python environment and installed by SlicerPytorch extension?

Reply:
When server is started from Slicer, SlicerPython will be used. If server is started outside of Slicer, a local environment is required.

Reply:
PyTorch is installed using the SlicerPytorch extension.

Reply:
Actually, let me check and I will get back to you

Reply:
@lassoan I made a few changes. When starting the server from Slicer, availability of PyTorch extension will be checked. If successful, other Python packages (monai) will be checked and installed.

As mentioned before, the server can be used independently from Slicer by just running uvicorn from the commandline.